test/classes/assignments/noAssignClass.chpl

```diff
@@ -0,0 +1,131 @@
+class A {
+  var x: int;
+}
+
+class B {
+  var x: int;
+}
+
+class C {
+  var x: int;
+}
+
+class D {
+  var x: int;
+}
+
+class E {
+  var x: int;
+}
+
+class F {
+  var x: int;
+}
+
+class G {
+  var x: int;
+}
+
+class H {
+  var x: int;
+}
+
+class I {
+  var x: int;
+}
+
+class J {
+  var x: int;
+}
+
+class K {
+  var x: int;
+}
+
+proc =(lhs: A, rhs: A) {
+  lhs.x = rhs.x;
+}
+
+proc =(lhs: unmanaged B, rhs: B) {
+  lhs.x = rhs.x;
+}
+
+proc =(lhs: borrowed C, rhs: C) {
+  lhs.x = rhs.x;
+}
+
+proc =(lhs: shared D, rhs: D) {
+  lhs.x = rhs.x;
+}
+
+proc =(lhs: owned E, rhs: E) {
+  lhs.x = rhs.x;
+}
+
+proc =(ref lhs: F, rhs: F) {
+  lhs.x = rhs.x;
+}
+
+proc =(ref lhs: owned G, rhs: G) {
+  lhs.x = rhs.x;
+}
+
+proc =(ref lhs: borrowed H, rhs: H) {
+  lhs.x = rhs.x;
+}
+
+proc =(ref lhs: I?, rhs: I) {
+  lhs.x = rhs.x;
+}
+
+proc =(ref lhs: owned J?, rhs: J) {
+  lhs.x = rhs.x;
+}
+
+proc =(ref lhs: borrowed K?, rhs: K) {
+  lhs.x = rhs.x;
+}
```

Comment:
It may be worth adding less-trivial assignment ops, such as class-from-int or class-from-record. I expect that things will work smoothly i.e. we get user errors, however it is good to check.

Reply:
The whitebox tester in me knows that these aren't necessary because my code only checks the first argument of the assignment overload, but the deferential reviewee in me has added them at your request.